### PR TITLE
Implement Entity Name filter

### PR DESCRIPTION
**Pull request by bytebuddha**

### CODE REVIEW

File: src/world_inspector/mod.rs [hunk 1]
```diff
@@ -174,13 +187,31 @@ impl<'a> WorldUIContext<'a> {
...
+                .body_returned
+                .unwrap_or(false)
+        } else if self.entity_name(entity).to_lowercase().contains(&params.name_filter.to_lowercase()) {
+            CollapsingHeader::new(self.entity_name(entity))
+                .id_source(id.with(entity))
+                .show(ui, |ui| {
+                    self.entity_ui_inner(ui, entity, params, id, entity_options)
+                })
+                .body_returned
+                .unwrap_or(false)
```
Comment (jakobhellermann) [inline src/world_inspector/mod.rs:205]:
```suggestion
        let entity_name = self.entity_name(entity);

        let show_entity = params.name_filter.is_empty()
            || params
                .name_filter
                .to_lowercase()
                .contains(&entity_name.to_lowercase());

        if show_entity {
            CollapsingHeader::new(self.entity_name(entity))
                .id_source(id.with(entity))
                .show(ui, |ui| {
                    self.entity_ui_inner(ui, entity, params, id, entity_options)
                })
                .body_returned
                .unwrap_or(false)
```

File: src/world_inspector/mod.rs [hunk 2]
```diff
@@ -45,6 +45,8 @@ pub struct WorldInspectorParams {
     pub despawnable_entities: bool,
     /// The window the inspector should be displayed on
     pub window: WindowId,
+    /// Filter entities by name.
+    pub name_filter: String
```
Comment (jakobhellermann) [inline src/world_inspector/mod.rs:49]: I think I'd rather have an `Option<String>` where `None` means that no filter UI will be shown. The default can be `Some(String::new())`.

**Review by jakobhellermann — CHANGES_REQUESTED** [review]

File: src/world_inspector/mod.rs (hunk 2 above)
Comment (bytebuddha) [reply src/world_inspector/mod.rs]: I think i have addressed most of your issues in the last 2 commit's. The only problem i see is if we are to use an Option<String> set to an empty String there really is no way to get back to `None` or vice versa, but one could easily add a system that toggle's that with a KeyCode or something if someone wants to be able to do that.

**Review by jakobhellermann — APPROVED** [review]

**Comment by jakobhellermann** [comment]
Looks good to me. Please run `cargo fmt` and I'll merge!

**Pull request merged**